### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Limits): generalize universes for preserving finite products

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Preserves/Finite.lean [hunk 1]
```diff
@@ -106,6 +107,13 @@ instance compPreservesFiniteProducts (F : C ⥤ D) (G : D ⥤ E)
 noncomputable instance (F : C ⥤ D) [PreservesFiniteLimits F] : PreservesFiniteProducts F where
   preserves _ _ := inferInstance
 
+noncomputable instance (priority := 100) (J : Type u) [Fintype J] (F : C ⥤ D)
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Limits/Preserves/Finite.lean:110]: Can it be `Finite J` instead?

Comment (dagurtomas) [reply Mathlib/CategoryTheory/Limits/Preserves/Finite.lean]: Yes! In fact we already had this for coproducts. I've changed it so products/coproducts are in sync.

**Review by TwoFX — COMMENTED** [review]
Thanks!
bors d+

File: Mathlib/CategoryTheory/Limits/Preserves/Finite.lean [hunk 2]
```diff
@@ -5,6 +5,7 @@ Authors: Andrew Yang
 -/
 import Mathlib.CategoryTheory.Limits.Preserves.Basic
 import Mathlib.CategoryTheory.Limits.Shapes.FiniteProducts
+import Mathlib.Data.Countable.Small
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Limits/Preserves/Finite.lean:8]:
```suggestion
```

**Comment by dagurtomas** [comment]
Thanks for the reviews!

bors merge

**Pull request closed without merging**